src/components/Searcher.js

```diff
@@ -970,6 +989,44 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     });
   }
 
+  /**
+   * Add signal each time a filter is added/removed.
+   * When a filter is added, signal with weight 1 is created.
+   * When a filter is removed, signal with weight 0 is created.
+   */
+  addFacetFilterSignal(facetFilter: FacetFilter, weight: number) {
+    const queryDocuments = this.state.response ? this.state.response.documents : null;
+    const querySignal = queryDocuments ? queryDocuments[0].signal : null;
```

Comment:
Just double checking... is it ever possible to not have any documents in the results when we get here? I don't think so, right? Any facet filters you can apply will always lead to having at least one document in the results I think.
In that case, we should be fine here.

Reply:
Yes, this is actually dependent on the signal of type click created. If that doesn't work for some reason, this would not too!

Reply:
If the user has somehow disabled metadata, than that's fine... we won't have signals for clicks and we won't have them for other things like facet filters.
I was just making sure that we always have at least one document in the search results...